.github/workflows/deploy.yml

```diff
@@ -1,17 +1,20 @@
-name: XCCup E2E Deployment
+name: XCCup Deployment
 on: workflow_dispatch
 jobs:
   build:
     name: Deploy
     runs-on: ubuntu-latest
     steps:
-      - name: executing remote ssh commands using password
+      - name: Login via SSH, pull changes from "main", build docker images and check afterwards the log
         uses: appleboy/ssh-action@master
         with:
           host: ${{ secrets.SSH_HOST }}
-          username: ${{ secrets.SSH_USERNAME }}
-          key: ${{ secrets.SSH_KEY }}
```

Comment:
Why not use the ssh keys? It's better to use them and we wanted to disable password ssh login anyway.